media/js/firefox/firstrun/firstrun.js

```diff
@@ -63,15 +66,42 @@
                     'interaction': 'fxa-login'
                 });
 
-                window.location.href = fxaIframeSrc + '/settings';
+                window.location.href = fxaIframeHost + '/settings';
 
                 break;
             }
         }
     }, true);
 
-    // load FxA iframe only after postMessage communication is configured
-    $fxaFrame.attr('src', $fxaFrame.data('src'));
+    // fallback in the event UITour fails for some reason
+    fxaIframeTimer = window.setTimeout(function() {
+        console.log('default FxA src being set...');
+        $fxaFrame.attr('src', fxaIframeSrc);
+        fxaIframeSrcSet = true;
+    }, 500);
+
+    // check user's Fx version to determine FxA iframe experience
+    Mozilla.UITour.getConfiguration('appinfo', function (config) {
+        console.log('uitour appinfo complete');
+        if (!fxaIframeSrcSet) {
+            // UITour succeeded - no need for fallback above
+            window.clearTimeout(fxaIframeTimer);
+
+            var version = parseInt(config.version, 10);
+
+            console.log('fx version = ' + version);
+
+            if (version >= 46) {
```

Comment:
I may well be missing something, but if you're only after the major version number then using UITour for this could be overkill perhaps? If you needed the minor version or channel name then UITour is the only way, but if the major version number is the only thing of importance, then I would probably use `Mozilla.Client.FirefoxMajorVersion` instead. This is both synchronous and does not require any special config.